# Capstone Project - The Battle of Neighborhoods
# Italian Restaurant in Toronto

## Introduction: Business Problem

Toronto is the most populous city in Canada, with a population of 2,731,571 in 2016. [1] Toronto is ethnically diverse and multicultural. Lots of immigrants with different backgrounds live in Toronto, and they bring their ethnic cuisine to this city such as Italian, Asian, Greek and etc. [2]
 
In this project, we explore the best locations for Italian restaurants in Toronto. Italian food is one of the most popular cuisine in the world. Just the North America pasta market reached a volume of nearly 3 Million Tons in 2018.[3] 

This report investigates which neighborhoods and boroughs of Toronto have the best and also highest number of Italian restaurants. Furthermore, this report will answer the questions *“Which Neighborhood have potential Italian restaurant Market?”,  “Where would I go in Toronto to have the best Italian food?”, and "Which areas dose not have any Italian restaurants"*


## Data

Following data sources will be needed to extract/generate the required information:


* Toronto data contains list of Boroughs, Neighborhoods and postcodes. We will use this data set to explore various neighborhoods of Toronto. Obtained from https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M


* Additional data including longitude and latitude for each postcode will be obtained from http://cocl.us/Geospatial_data 
  

* All data related to locations and quality of Italian restaurants in Toronto will be obtained via the **Foursquare API** 


# Methodology

The first step is to get required data about Toronto, which is available on Wikipedia. By using BeautifulSoup, the data  will be extracted from wikipedia and then transfered to dataframe. Next, additional data including post Code, longitude and latitude will be obtained from http://cocl.us/Geospatial_data and transfered to another dataframe. Eventually two dataframes will be attached at the mutual column.

By using FourSquare API, all venues for each Neighborhood can be transfered to dataframe. By filtering these data, Italian restaurants will be listed. In addition, the count of tips and likes, and rating is obtained by FourSquare API and transfered to dataframe. We will have the average rating for each Neighbourhood and borough, and visualize the ranking of neighborhoods using folium library.

# Results & Discussion


The following table includes the dataset of 103 Neigborhoods in Toronto. The largest number of Neigborhoods in each Borough is in North York

In [17]:
import pandas as pd
toronto_data_csv=pd.read_csv('toronto_data_csv.csv')
toronto_data_csv.shape

(103, 4)

In [18]:
toronto_data_csv.head()

Borough                                 Neighborhood   Latitude  \
0        North York                                    Parkwoods  43.753259   
1        North York                             Victoria Village  43.725882   
2  Downtown Toronto                    Regent Park, Harbourfront  43.654260   
3        North York             Lawrence Manor, Lawrence Heights  43.718518   
4  Downtown Toronto  Queen's Park, Ontario Provincial Government  43.662301   

   Longitude  
0 -79.329656  
1 -79.315572  
2 -79.360636  
3 -79.464763  
4 -79.389494

<img src= "NB.png" style="width:900px;height:600px"/>

According to API data, there are 105 Italian restaurants in Toronto. In the following table, you can see the name of Italian restaurants in each Neighborhood and Borough.

In [19]:
italian_rest_t_csv=pd.read_csv('italian_rest_t_csv.csv')
italian_rest_t_csv.shape

(105, 4)

In [20]:
italian_rest_t_csv.head()

Borough                                  Neighborhood  \
0  Downtown Toronto                    Regent Park / Harbourfront   
1  Downtown Toronto                    Regent Park / Harbourfront   
2  Downtown Toronto                    Regent Park / Harbourfront   
3  Downtown Toronto  Queen's Park / Ontario Provincial Government   
4  Downtown Toronto  Queen's Park / Ontario Provincial Government   

                         ID                          Name  
0  4ad776eef964a520e20a21e3     Mangia and Bevi Resto-Bar  
1  4cbdc6784495721ea262617a                      Fusaro's  
2  56d8dff7498eb4e5e661e78d                          Ardo  
3  4a8355bff964a520d3fa1fe3                      Mercatto  
4  5a69ba9112138476a91996f8  Dal Moro's Fresh Pasta To Go

<img src= "BR.png" style="width:900px;height:600px"/>

Davisville in Central Toronto has the largest number (eight) of Italian restaurants.

<img src= "NR.png" style="width:900px;height:600px"/>

Then we need further information such as likes, rating and tips for each restaurant for additional analysis. The data will be transfered into dataframe and saved as csv file. 

In [16]:
italian_details_csv=pd.read_csv('italian_details_csv.csv')
italian_details_csv.head(6)

Borough                                  Neighborhood  \
0  Downtown Toronto                    Regent Park / Harbourfront   
1  Downtown Toronto                    Regent Park / Harbourfront   
2  Downtown Toronto                    Regent Park / Harbourfront   
3  Downtown Toronto  Queen's Park / Ontario Provincial Government   
4  Downtown Toronto  Queen's Park / Ontario Provincial Government   
5  Downtown Toronto                     Garden District / Ryerson   

                         ID                          Name  Likes  Rating  Tips  
0  4ad776eef964a520e20a21e3     Mangia and Bevi Resto-Bar     77     8.4    40  
1  4cbdc6784495721ea262617a                      Fusaro's     57     8.5    22  
2  56d8dff7498eb4e5e661e78d                          Ardo     15     7.4     8  
3  4a8355bff964a520d3fa1fe3                      Mercatto     56     7.8    26  
4  5a69ba9112138476a91996f8  Dal Moro's Fresh Pasta To Go      9     8.2     6  
5  4b49183ff964a520a46526e3                       Terroni    278     8.1    95

Based on  above table, we can find out the following information:  
1- The most number of likes (275 times) belongs to Terroni in  Garden District / Ryerson in     Downtown Toronto   
2- The highest rating Italian restaurant is Nodo in High Park / The Junction South in West Toronto (rating=9).   
3- The maximum tips belongs to  Terroni in Garden District / Ryerson in Downtown Toronto with 95 tips.   

All restaurants are grouped by Neighborhood, and showed the average Rating for each Neighborhood. Italian restaurants in New Toronto / Mimico South / Humber Bay Shores has the highest average Rating (8.7/10)

Next all restaurants are grouped by Borough, and showed the average Rating for each Borough. Italian restaurants in Downtown Toronto has the highest average Rating (8.1/10)

<img src= "BRA.png" style="width:900px;height:600px"/>

The Neighborhood with average rating of 8.5 or higher is filtered, and merged to Toronto dataframe. By means of merged dataframe and Folium library, we can see the two Neighborhoods with highest average Rating.

<img src= "table.png" style="width:900px;height:600px"/>

<img src= "Screenshot.png">

# Conclusion

Davisville (Downtown Toronto) has the most number of italian restaurants in Toronto. New Toronto (Etobicoke), St. James Town(Downtown Toronto), Little Portugal (West Toronto) are some of the best neighborhoods for Italian cuisine in Toronto. Downtown Toronto have potential Italian Resturant Market, there are multiple neighborhoods in Downtown Toronto with average ratings exceeding 8.1. East Park ranks last in average rating (0 from 10) of Italian restaurants. It might be as a result of limitation of data providing only by Foursquare. In addition, I would go to Terroni in Downtown Toronto for the best Italian food based on 278 likes and Rating of 8.1. Downtown Toronto and West Toronto are the best place to stay if you prefer Italian Cuisine.

# Rerefrences
[1] https://en.wikipedia.org/wiki/Toronto   
[2] https://en.wikipedia.org/wiki/Cuisine_in_Toronto   
[3] https://www.imarcgroup.com/north-america-pasta-market   